src/Component.js

```diff
@@ -0,0 +1,62 @@
+
+var __extends = (this && this.__extends) || function (d, b) {
```

Comment:
That is the standard extending function that compiled typescript JS uses, so I used it because I know it's well tested in many environments. But I'm not tied to it. Any function that mimics a ES6 class style `extends` will do fine, since this is just an ES5 adaptation of an ES6 class I had been using for these features.

I avoided `Object.assign` because it doesn't have the level of browser support needed to just use in end-product code like this unless reflux includes a polyfill that I don't know about. And frankly...I just didn't know reflux core had an extending utility in it.